app/models/oidc/trusted_publisher/buildkite.rb

```diff
@@ -0,0 +1,127 @@
+class OIDC::TrustedPublisher::Buildkite < ApplicationRecord
+  has_many :rubygem_trusted_publishers, class_name: "OIDC::RubygemTrustedPublisher", as: :trusted_publisher, dependent: :destroy,
+    inverse_of: :trusted_publisher
+  has_many :pending_trusted_publishers, class_name: "OIDC::PendingTrustedPublisher", as: :trusted_publisher, dependent: :destroy,
+    inverse_of: :trusted_publisher
+  has_many :rubygems, through: :rubygem_trusted_publishers
+  has_many :api_keys, dependent: :destroy, inverse_of: :owner, as: :owner
+
+  validates :organization_slug, :pipeline_slug,
+    presence: true, length: { maximum: Gemcutter::MAX_FIELD_LENGTH }
+
+  validate :unique_publisher
+
+  def self.for_claims(claims)
+    organization_slug = claims.fetch(:organization_slug)
+    pipeline_slug = claims.fetch(:pipeline_slug)
+
+    where(organization_slug:, pipeline_slug:).first!
+  end
+
+  def self.permitted_attributes
+    %i[organization_slug pipeline_slug]
+  end
+
+  def self.build_trusted_publisher(params)
+    params = params.reverse_merge(organization_slug: nil, pipeline_slug: nil)
+    find_or_initialize_by(params)
+  end
+
+  def self.publisher_name = "Buildkite"
+
+  def payload
+    {
+      name:,
+      organization_slug:,
+      pipeline_slug:
+    }
+  end
+
+  delegate :as_json, to: :payload
+
+  def organization_slug_condition
+    OIDC::AccessPolicy::Statement::Condition.new(
+      operator: "string_equals",
+      claim: "organization_slug",
+      value: organization_slug
+    )
+  end
+
+  def pipeline_slug_condition
+    OIDC::AccessPolicy::Statement::Condition.new(
+      operator: "string_equals",
+      claim: "pipeline_slug",
+      value: pipeline_slug
+    )
+  end
+
+  def audience_condition
+    OIDC::AccessPolicy::Statement::Condition.new(
+      operator: "string_equals",
+      claim: "aud",
+      value: Gemcutter::HOST
+    )
+  end
+
+  def to_access_policy(jwt)
+    # TODO what to do with jwt here?
```

Comment:
I think we're OK to ignore it, since there are no cross-claim consistency checks we need to enforce, unlike with GitHub (where we make sure the workflow source matches up with the commit being built)